lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -919,8 +913,8 @@ void onAttached(boolean hasPresence) {
     /**
      * Spec: RTP17g
      */
-    synchronized void enterInternalMembers() {
-        for (final PresenceMessage item: internalPresence.values()) {
+    void enterInternalMembers() {
```

Comment:
Could you explain why we deleted `synchronized` here?

Reply:
This was added as a part of `no-connection-serial`. Actually, there was no need to add `synchronized` for enteringPresenceMembers. Sometimes, this was causing intermittent waits  for threads trying to call `enteringPresenceMembers`. We don't need thread synchronization here. We are doing the same in ably-go -> https://github.com/ably/ably-go/blob/dd8236fc301eb078803e83efee7615f3ba133341/ably/realtime_presence.go#L187

Reply:
This was added as a part of no-connection-serial ( probably referred to some old code that had synchronized marked as true ). But, we don't actually need it here.